storage/bootstrap/bootstrapper/fs/source.go

```diff
@@ -116,11 +116,7 @@ func (s *fileSystemSource) enqueueReaders(
 	readersCh chan<- shardReaders,
 ) {
 	for shard, tr := range shardsTimeRanges {
-		var files []string
-		fs.ForEachInfoFile(s.filePathPrefix, namespace, shard, s.readerBufferSize, func(fname string, _ []byte) {
-			files = append(files, fname)
-		})
-
+		files := fs.ReadInfoFiles(s.filePathPrefix, namespace, shard, s.readerBufferSize)
```

Comment:
@r yep that was the intent of this change. The file `source.go` is the only non-package caller of the persist/fs ReadInfo* calls. This change makes the two calls use the `ReadInfoFiles` method.

@xichen2020 `source.go` earlier called `ForEachInfoFile` and `ReadInfoFiles` both. Now it calls `ReadInfoFiles` in both places. In turn, `ReadInfoFiles` internally calls `ForEachInfoFile` - [1] (renamed to `forEachInfoFile`), so it still performs all the sanity checking you're asking about - [2]. It performs one additional check compared to earlier - the readInfo call in the internal func definition.

[1]: https://github.com/m3db/m3db/blob/9f335641e25380c73eb6b8dc61c320738f1779ad/persist/fs/files.go#L209-L214
[2]: https://github.com/m3db/m3db/blob/9f335641e25380c73eb6b8dc61c320738f1779ad/persist/fs/files.go#L177-L200